socorro/external/crashstorage_base.py

```diff
@@ -630,26 +589,33 @@ def save_raw_and_processed(self, raw_crash, dump, processed_crash,
         # processed crash which is a SocorroDotDict into a pure python
         # dict which we can more easily copy.deepcopy() operate on.
         processed_crash_as_dict = socorrodotdict_to_dict(processed_crash)
+        raw_crash_as_dict = socorrodotdict_to_dict(raw_crash)
 
         for a_store in self.stores.itervalues():
             self.quit_check()
             try:
                 actual_store = getattr(a_store, 'wrapped_object', a_store)
 
-                if hasattr(actual_store, 'is_mutator') and actual_store.is_mutator():
+                if (
+                    hasattr(actual_store, 'is_mutator') and
+                    actual_store.is_mutator()
+                ):
                     # We do this because `a_store.save_raw_and_processed`
                     # expects the processed crash to be a DotDict but
                     # you can't deepcopy those, so we deepcopy the
                     # pure dict version and then dress it back up as a
                     # DotDict.
-                    crash = SocorroDotDict(copy.deepcopy(processed_crash_as_dict))
-                else:
-                    crash = processed_crash
+                    processed_crash = SocorroDotDict(
+                        copy.deepcopy(processed_crash_as_dict)
+                    )
+                    raw_crash = SocorroDotDict(
+                        copy.deepcopy(raw_crash_as_dict)
+                    )
```

Comment:
`processed_crash` and `raw_crash` are passed in as arguments to this function. So now the values are sometimes the original values and sometimes copies. That's messy and I claim it makes things harder to debug. I think we want to avoid re-using those variable names in this way.

I'd change this to:

```
    my_processed_crash = SocorroDotDict(copy.deepcopy(processed_crash_as_dict))
    my_raw_crash = SocorroDotDict(copy.deepcopy(raw_crash_as_dict))
else:
    my_processed_crash = processed_crash
    my_raw_crash = raw_crash
```

Then use `my_raw_crash` and `my_processed_crash` in the `.save_raw_and_processed()` call below.